### PR TITLE
test(components): Added integration test for Sagemaker TrainingJob component v2.

### PR DESCRIPTION
**Pull request by ananth102**
**Description of your changes:**
Adds 2 integration tests for the Trainingjob v2 component. Also incudes installation of [ACK](https://aws-controllers-k8s.github.io/community/docs/tutorials/sagemaker-example/).

List of changes:
- `components/aws/sagemaker/tests/integration_tests/Dockerfile` - Installs helm and yq for ack installation.
- `components/aws/sagemaker/tests/integration_tests/component_tests/test_v2_train_component.py` - tests for TrainingJob v2.
- `components/aws/sagemaker/tests/integration_tests/conftest.py` - Kubernetes client fixture.
- `components/aws/sagemaker/tests/integration_tests/resources/config/ack-training-job/config.yaml` - Configuration for test.
- `components/aws/sagemaker/tests/integration_tests/scripts/generate_trust_policy` - Added permissions for ack controller.
- `components/aws/sagemaker/tests/integration_tests/scripts/run_integration_tests` - Installs/uninstalls ack controller.
- `components/aws/sagemaker/tests/integration_tests/utils/ack_utils.py` - Helper function for polling ack.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: components/aws/sagemaker/tests/integration_tests/scripts/run_integration_tests [hunk 1]
```diff
@@ -164,6 +168,43 @@ function install_kfp() {
   echo "[Installing KFP] Pipeline pods are ready"
 }
 
+function install_ack(){
+
+  eksctl utils associate-iam-oidc-provider --cluster ${EKS_CLUSTER_NAME} \
```
Comment (surajkota) [inline components/aws/sagemaker/tests/integration_tests/scripts/run_integration_tests:173]: this is already done in `generate_iam_role`
https://github.com/kubeflow/pipelines/blob/master/components/aws/sagemaker/tests/integration_tests/scripts/generate_iam_role#L85

File: components/aws/sagemaker/tests/integration_tests/conftest.py [hunk 2]
```diff
@@ -148,6 +149,9 @@ def kfp_client():
     kfp_installed_namespace = utils.get_kfp_namespace()
     return kfp.Client(namespace=kfp_installed_namespace)
 
+@pytest.fixture(scope="session")
+def k8s_client():
```
Comment (surajkota) [inline components/aws/sagemaker/tests/integration_tests/conftest.py:153]: its better to create a new session everytime see https://github.com/aws-controllers-k8s/test-infra/blob/main/src/acktest/k8s/resource.py#L119-L122

optional: update k8s version https://github.com/kubeflow/pipelines/blob/master/components/aws/sagemaker/tests/integration_tests/environment.yml#L16. Should we update all dependencies to latest in here and switch python version to 3.8?

**Review by surajkota — CHANGES_REQUESTED** [review]
Nice work! Few comments. Feel free to merge once addressed

File: components/aws/sagemaker/tests/integration_tests/conftest.py (hunk 2 above)
Comment (ananth102) [reply components/aws/sagemaker/tests/integration_tests/conftest.py:153]: The old components rely on older boto3/sagemaker versions so its probably not possible.

Comment (surajkota) [reply components/aws/sagemaker/tests/integration_tests/conftest.py]: These are just test dependencies, not directly related to source code

**Comment by surajkota** [comment]
Run black to lint the code

**Comment by ananth102** [comment]
> Run black to lint the code

Kfp_client times out around 20 minutes, with black formatting one of the tests fails because it starts after 20 minutes.

File: components/aws/sagemaker/tests/integration_tests/component_tests/test_v2_train_component.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        download_dir,
+        test_params["TestName"],
+        60,
+        "running",
+    )
+    print(f"Terminating run: {run_id} where Training job_name: {input_job_name}")
+    kfp_client_utils.terminate_run(kfp_client, run_id)
+    # response = ack_utils.describe_training_job(k8s_client, input_job_name)
+    desiredStatuses = ["Stopping", "Stopped"]
+    training_status_reached = ack_utils.wait_for_trainingjob_status(k8s_client,input_job_name,desiredStatuses,10,6)
```
Comment (surajkota) [inline components/aws/sagemaker/tests/integration_tests/component_tests/test_v2_train_component.py:93]: have you run linting on this file? Iwas expecting this to become `(k8s_client, input_job_name, desiredStatuses, 10, 6)`

File: components/aws/sagemaker/tests/integration_tests/component_tests/test_v2_train_component.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        test_params["PipelineDefinition"],
+        test_params["Arguments"],
+        download_dir,
+        test_params["TestName"],
+        60,
+        "running",
+    )
+    print(f"Terminating run: {run_id} where Training job_name: {input_job_name}")
+    kfp_client_utils.terminate_run(kfp_client, run_id)
+    # response = ack_utils.describe_training_job(k8s_client, input_job_name)
```
Comment (surajkota) [inline components/aws/sagemaker/tests/integration_tests/component_tests/test_v2_train_component.py:91]: nit: remove

**Review by surajkota — APPROVED** [review]
/approve

File: components/aws/sagemaker/tests/integration_tests/component_tests/test_v2_train_component.py (hunk 3 above)
Comment (ananth102) [reply components/aws/sagemaker/tests/integration_tests/component_tests/test_v2_train_component.py]: done, codebuild works as well

**Comment by chensun** [comment]
Test failure is unrelated to the PR.

**Pull request merged**